### ISSUE TITLE
[Legacy Upgrade] 2.1.1 x64 and x86 MSI installers retains legacy x86 Start Menu entries

### ISSUE THREAD

**Issue by Thibi** [windirstat/windirstat#155]
Version: 2.1.1
Legacy version: 1.1.2 x86
Impacted installers:

- WinDirStat-x64.msi
- WinDirStat-x86.msi

Legacy impacted installer: windirstat1_1_2_setup.exe
OS: Win11 23H2

Impact scope: Upgrades

- The ARM installers were unable to be tested.
- I unchecked the option to create a classic context menu entry.

Upon upgrade installation it leaves the old v1.1.2 Start Menu entries and Start Menu folder that point to:

- WinDirStat | "C:\Program Files (x86)\WinDirStat\WinDirStat.exe"
- Uninstall WinDirStat | "C:\Program Files (x86)\WinDirStat\Uninstall.exe" _(2.1.1 x86 installer removes this shortcut during installation. It may be a legacy item that appears with Win11 tomfoolery)_
- Help (ENG) | "C:\Program Files (x86)\WinDirStat\windirstat.chm"

It still installs the correct 2.1.1 Start Menu entries.

If 1.1.2 is uninstalled directly _then_ 2.1.1 is installed, the entries are correct. 1.1.2 removes all of its own files correctly upon uninstallation.

**Comment by Thibi** [comment]
This may be due to upgrading from version 1.1.2?

I had uninstalled 2.1.1 x64 and reinstalled it then uninstalled it again to test the 2.1.1 x86 installer. Uninstalling that then reinstalling the 2.1.1 x64 provided the correct Start Menu Entry.

**Comment by NoMoreFood** [comment]
Yes, this is due to me not removing the old Start Menu icons properly. I have this corrected in the 2.1.2 beta version (released soon).

**Comment by Neustradamus** [comment]
@Thibi: Please look it:
- https://github.com/windirstat/windirstat/issues/145#issuecomment-2516389206
- https://github.com/windirstat/windirstat/issues/145#issuecomment-2526277681
- https://github.com/windirstat/windirstat/issues/149

cc: @NoMoreFood.

**Comment by Neustradamus** [comment]
I have installed the https://github.com/windirstat/windirstat/releases/tag/beta%2Fv2.1.2%2F2024-12-12 version after 2.1.1, the shortcuts are always here :/

**Comment by NoMoreFood** [comment]
@Neustradamus Did you reinstall the legacy version prior to that?  It only removes them if it detects the presence of the legacy registry keys.

**Comment by NoMoreFood** [comment]
I'll consider adding unconditional logic for cleaning up the icons.

**Comment by NoMoreFood** [comment]
Updated to always cleanup in: 

https://github.com/windirstat/windirstat/releases/tag/beta/v2.1.2/2024-12-17

Please let me know if you still see any remnants.

**Comment by Neustradamus** [comment]
Here:
- https://github.com/windirstat/windirstat/commit/e412f1c2cac891830c6913a74c3b1d14d8e5aa1b

You can see Installer improvements here:
- https://github.com/windirstat/windirstat/issues/149

**Comment by NoMoreFood** [comment]
Released with https://github.com/windirstat/windirstat/releases/tag/release/v2.2.0

If any additional action is warranted, please submit a new issue.
